src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIRemoteProjectService.java

```diff
@@ -0,0 +1,127 @@
+
+/**
+ * A utility class for doing operations on remote project sync settings.
+ */
+
+@Component
+public class UIRemoteProjectService {
+	private MessageSource messageSource;
+	private ProjectService projectService;
+	private ProjectRemoteService projectRemoteService;
+	private UserService userService;
+
+	@Autowired
+	public UIRemoteProjectService(MessageSource messageSource, ProjectService projectService,
+			ProjectRemoteService projectRemoteService, UserService userService) {
+		this.messageSource = messageSource;
+		this.projectService = projectService;
+		this.projectRemoteService = projectRemoteService;
+		this.userService = userService;
+	}
+
+	/**
+	 * Update the project sync settings
+	 *
+	 * @param projectId                          the project id to update
+	 * @param remoteProjectSettingsUpdateRequest object which
+	 *                                           is used to update frequency, and sync user, as well as
+	 *                                           force sync for a project
+	 * @param principal                          The current logged in user
+	 * @param locale                             user's locale
+	 * @return {@link AjaxResponse}
+	 */
+	public AjaxResponse updateProjectSyncSettings(Long projectId,
+			RemoteProjectSettingsUpdateRequest remoteProjectSettingsUpdateRequest, Principal principal, Locale locale) {
+		Project project = projectService.read(projectId);
+
+		if (project != null) {
+			RemoteStatus remoteStatus = project.getRemoteStatus();
+			Map<String, Object> updates = new HashMap<>();
+
+			String message = null;
+			String error = null;
+
+			if (remoteProjectSettingsUpdateRequest.getProjectSyncFrequency() != null) {
+				updates.put("syncFrequency", remoteProjectSettingsUpdateRequest.getProjectSyncFrequency());
+				message = messageSource.getMessage("server.ProjectRemote.frequency.change",
+						new Object[] { remoteProjectSettingsUpdateRequest.getProjectSyncFrequency() }, locale);
+			}
+
+			if (remoteProjectSettingsUpdateRequest.getForceSync()) {
+				remoteStatus.setSyncStatus(RemoteStatus.SyncStatus.MARKED);
+				updates.put("remoteStatus", remoteStatus);
+				message = messageSource.getMessage("server.ProjectRemote.marked.for.sync", new Object[] {}, locale);
+			}
+
+			if (remoteProjectSettingsUpdateRequest.getChangeUser()) {
+				// ensure the user can read the project
+				try {
+					projectRemoteService.read(remoteStatus.getURL());
+
+					User user = userService.getUserByUsername(principal.getName());
+					remoteStatus.setReadBy(user);
+					updates.put("remoteStatus", remoteStatus);
+
+					message = messageSource.getMessage("server.ProjectRemote.userchange", new Object[] {}, locale);
+				} catch (Exception ex) {
+					error = messageSource.getMessage("server.ProjectRemote.userchange.error", new Object[] {}, locale);
+				}
+			}
+
+			projectService.updateProjectSettings(project, updates);
+
+			if (error != null) {
```

Comment:
Use `Strings.`_`isNullOrEmpty`_`(error)` for these checks

Reply:
Removed in dc6a7340bb060bcf798d856d3e29b7da3fff9389 as we are catching the exception in the controller and throws the exception and returns the message